From: Ben Dooks <ben.dooks@codethink.co.uk>
Subject: devres: show which resource was invalid in __devm_ioremap_resource()
Date: Wed, 21 Jun 2023 17:30:50 +0100

The other error prints in this call show the resource which wsan't valid,
so add this to the first print when it checks for basic validity of the
resource.

Link: https://lkml.kernel.org/r/20230621163050.477668-1-ben.dooks@codethink.co.uk
Signed-off-by: Ben Dooks <ben.dooks@codethink.co.uk>
Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/devres.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/devres.c~devres-show-which-resource-was-invalid-in-__devm_ioremap_resource
+++ a/lib/devres.c
@@ -129,7 +129,7 @@ __devm_ioremap_resource(struct device *d
 	BUG_ON(!dev);
 
 	if (!res || resource_type(res) != IORESOURCE_MEM) {
-		dev_err(dev, "invalid resource\n");
+		dev_err(dev, "invalid resource %pR\n", res);
 		return IOMEM_ERR_PTR(-EINVAL);
 	}
 
_
